packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -137,75 +143,26 @@ abstract class VirtualNodeBase extends cdk.Resource implements IVirtualNode {
   }
 
   /**
-   * Utility method to add an inbound listener for this virtual node
+   * Utility method to add an inbound listener for this VirtualNode
    */
   public addListeners(...listeners: VirtualNodeListener[]) {
-    if (this.listeners.length + listeners.length > 1) {
+    if (listeners.length + this.listeners.length > 1) {
       throw new Error('VirtualNode may have at most one listener');
     }
```

Comment:
We discussed this offline, but wanted to put it here.

We are going to accept the array of listeners without validation. This is to future proof against when we lift the server side validation for a single listener. We do not want to require customers to upgrade their CDK to use multiple listeners